tests/unit/test_mediatool.py

```diff
@@ -0,0 +1,156 @@
+    def test_get_campaigns(self, mock_handle_api_response):
+        """Test Mediatool `_get_campaigns` function."""
+        mock_response = MagicMock()
+        mock_response.text = json.dumps(variables["campaigns"])
+        mock_handle_api_response.return_value = mock_response
+
+        result = self.mediatool._get_campaigns(organization_id="org_id")
+        expected_result = [{"_id": "1", "name": "Campaign1"}]
+        self.assertEqual(result, expected_result)
+
+    @pytest.mark.functions()
+    @patch("viadot.sources.mediatool.handle_api_response")
+    def test_get_media_types(self, mock_handle_api_response):
+        """Test Mediatool `_get_media_types` function."""
+        mock_response = MagicMock()
+        mock_response.text = json.dumps(variables["media_types"])
+        mock_handle_api_response.return_value = mock_response
+
+        result = self.mediatool._get_media_types(media_type_ids=["1"])
+        expected_result = [{"_id": "1", "name": "Type1", "type": "Type"}]
+        self.assertEqual(result, expected_result)
+
+    @pytest.mark.functions()
+    def test_rename_columns(self):
+        """Test Mediatool `_rename_columns` function."""
+        df = pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
+        result = self.mediatool._rename_columns(df, column_suffix="test")
+        expected_result = pd.DataFrame({"col1_test": [1, 2], "col2_test": [3, 4]})
+        pd.testing.assert_frame_equal(result, expected_result)
+
+    @pytest.mark.connect()
+    @patch("viadot.sources.mediatool.handle_api_response")
+    def test_api_connection(self, mock_handle_api_response):
+        """Test Mediatool `api_connection` method."""
+        mock_response = MagicMock()
+        mock_response.text = json.dumps(variables["organizations"])
+        mock_handle_api_response.return_value = mock_response
+
+        result = self.mediatool.api_connection(get_data_from="organizations")
+        expected_result = [{"_id": "1", "name": "Org1", "abbreviation": "O1"}]
+        self.assertEqual(result, expected_result)
+
+    @pytest.mark.functions()
+    @patch("viadot.sources.mediatool.handle_api_response")
+    def test_to_df(self, mock_handle_api_response):
+        """Test Mediatool `to_df` method."""
+        mock_response = MagicMock()
+        mock_response.text = json.dumps(
+            {"mediaEntries": [{"_id": "1", "name": "Entry1"}]}
+        )
+        mock_handle_api_response.return_value = mock_response
+
+        data = [{"_id": "1", "name": "Entry1"}]
+        result_df = self.mediatool.to_df(data=data, column_suffix="media_entries")
+        result_df.drop(
+            columns=["_viadot_source", "_viadot_downloaded_at_utc"],
+            inplace=True,
+            axis=1,
+        )
+        expected_result = pd.DataFrame(
+            {"_id_media_entries": ["1"], "name_media_entries": ["Entry1"]}
+        )
+        pd.testing.assert_frame_equal(result_df, expected_result)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
Is this required when running this with pytest?

Reply:
changed to avoid that structure.